Feature/main page 메인 페이지 수정

2NNS-V opened the pull request:
## 구현한 기능

- 메인 페이지에서 img 추가
- 드롭다운 컴포넌트 추가 및 적용
- 통계 페이지 퍼블리싱
- 전역 변수 설정
- 동아리 로고 Footer 컴포넌트 추가
- png -> svg 확장자명 변경

## 논의하고 싶은 내용

- 

## 기타

- 통계페이지 디자인은 아직 수정중

2NNS-V committed:
feat:통계 페이지로 이동

2NNS-V committed:
Merge branch 'develop' of https://github.com/LikeLion-KNU/dongBTI into feature/main-page

2NNS-V committed:
design:백그라운드 이미지 추가

2NNS-V committed:
design:사이드바 hover 이벤트 추가

2NNS-V committed:
design:input 폰트 크기 설정 및 padding 설정

2NNS-V committed:
feat:단과대학 dropdown 추가 및 적용

2NNS-V committed:
design:background image 삭제

2NNS-V committed:
chore: png->svg로 확장자명 변경

2NNS-V committed:
style:sidebar 블러 효과 수정

2NNS-V committed:
feat: 변경된 디자인에 맞게 수정 및 전역 상태(name, major) 설정

2NNS-V committed:
style: input 글씨체 설정

2NNS-V committed:
refact: 이름 명시적으로 수정

2NNS-V committed:
feat:통계 페이지 진행중

2NNS-V committed:
feat:메인 페이지에 이미지 추가

2NNS-V committed:
feat:동아리 로고 Footer 추가

2NNS-V committed:
feat:Footer 적용 및 메인 이미지 반응형 적용

toothlessdev commented on src/components/display/Footer.tsx:
```diff
@@ -0,0 +1,24 @@
+    );
+}
+
+const FooterWrapper = styled.div`
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    gap: 10px;
+    margin: 20px;
+`;
```
이부분 `Footer.style.ts` 로 스타일 관련 코드를 분리하는게 좋아 보입니다!

toothlessdev commented on src/components/form/Dropdown.tsx:
```diff
@@ -0,0 +1,49 @@
+    setSelectedMajor: React.Dispatch<React.SetStateAction<string>>;
+}
+
+export default function dropdown({ color, width, height, selectedMajor, setSelectedMajor }: DropdownProps) {
```
컴포넌트 이름은 CamelCase 로 작성해주세요!

toothlessdev replied:
`export defualt` 로 내보냈기 때문에 `import DropDown from "@/components/form/DropDown"` 으로 참조 가능하지만, HTML 기본 DOM Element 와 구별하기 위해 React 공식문서에서도 CamelCase 로 작성하는것을 권장하고 있습니다

2NNS-V replied:
`dropDown`이 camelCase로 알고 있는데, `DropDown`으로 써야 하나요?

toothlessdev replied:
아 PascalCase 입니다! 헷갈렸네요;; `DropDown` 으로 변경해주시면 될것 같습니다!

toothlessdev commented on src/components/navigation/SideBar.tsx:
```diff
@@ -28,18 +30,21 @@ export const SideBar = ({isOpen, setIsOpen} : SideBarProps) => {
 
-  return (
-    
+    const handleNavigation = (path: string) => {
+        navigate(path);
+        handleSideBar(); 
+    };
+
+  return ( 
```
path 매개변수에 analytics 이외의 값이 들어가지 않는 다면 다른 path 를 받지 않고 바로 `navigate("/analytics")` 와 같이 작성하는게 좋지 않을까요 ?
handleSideBar 또한 그냥 `setIsOpen(false)` 를 호출해도 상관 없을것 같아요!

예)
```tsx
const navigateToAnalytics = () => {
    navigate("/analytics");
    setIsOpen(false);
}

<Menu onClick={navigateToAnalytics}>
```

2NNS-V replied:
```
           <ul>
                <Menu>테스트</Menu>
                <Menu onClick={() => handleNavigation("/analytics")}>통계</Menu>
                <Menu>만든이들</Menu>
            </ul>
```
각 메뉴마다 이동할 때 `handleNavigation("/select")`, `handleNavigation("/makers")` 이런 식으로 세번 사용될 것 같아요!

2NNS-V replied:
handleNavigation 함수명을 직관적으로 바꾸는게 좋을까요?

toothlessdev replied:
아 이해했습니다! 아직 테스트와 만든이들에 연결이 안되어있군요! 그대로 두셔도 괜찮을것 같아요!

2NNS-V committed:
refact: Footer 스타일 컴포넌트 분리

2NNS-V committed:
refact: dropdown 컴포넌트명 PascalCase로 변경

2NNS-V committed:
refact: 불필요한 함수 제거(handleSideBar)

2NNS-V committed:
feat:SideBar 이동 로직 추가

toothlessdev reviewed (approved):
LGTM! any 는 최대한 피하고 명시적으로 타입을 지정하는 습관을 들이면 좋을것 같아요!

2NNS-V committed:
Rename:Dropdown.tsx to DropDown.tsx

2NNS-V committed:
Rename:Dropdown.style.ts to DropDown.style.ts

2NNS-V committed:
refact: DropDown 코드 리팩토링

- LabelContainer 타입 명시
- Label과 DropDown에서 공통되는 Background color 로직 추출
- DropDownProps 하나로 처리

2NNS-V committed:
refact: SideBar 관련 함수 타입 지정

toothlessdev reviewed (approved):
LGTM!
